### ISSUE TITLE
rename EncodedValues to RoadAttributes

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#2449]
And change documentation (like for custom models) accordingly.

Plus maybe change PathDetail to RoadAttribute too ... although this does not fit 100% as there are path details like edge_id or street_name that won't be implemented via RoadAttribute.

**Comment by michaz** [comment]
Is it a RoadAttribute, or maybe an EdgeAttribute?

**Comment by easbar** [comment]
Yes essentially they are road attributes that are assigned to edges. So after the import they are attributes of edges in a way. I like `RoadAttribute`, because end-users will also deal with these things (in custom models or path details) and they rather think in terms of 'roads' than 'edges'. It doesn't seem worth distinguishing the different levels of abstraction here? Then again if we use the `BaseGraph` for different things than the road network (like the time-expanded PT graph) `RoadAttribute` probably doesn't make much sense from this point of view. But on the other hand there are already multiple aspects of `BaseGraph` that make it a `RoadNetwork` rather than a general purpose graph structure (like way geometry for example). Btw I don't see much value in creating a general purpose graph structure.

**Comment by michaz** [comment]
I'm only saying, we are *in fact* calling them Edges right now. As in EdgeIterator, getEdge etc.

**Comment by easbar** [comment]
Ok, I would totally call them `EdgeAttributes` just like `EdgeIterator` etc. The (only) reason I would call them `RoadAttributes` is that this concept bubbles up all the way to end-users (even in Directions API docs) where otherwise no understanding of graph concepts (like edges) is/should be required (at least I thought so so far). So we either need to confront users with a terminology they might not be familiar with (like 'edge' or (much worse) 'encoded value'), or we distinguish this somehow (call it `EdgeAttribute` internally and `RoadAttribute` on a higher level), or we call it `RoadAttribute` and live with the fact that the better internal name would have been `EdgeAttribute`.
